src/statsd_client.cpp

```diff
@@ -76,20 +77,14 @@ struct _StatsdClientData {
 StatsdClient::StatsdClient(const string& host, int port, const string& ns)
 {
     d = new _StatsdClientData;
-    d->sock = -1;
+    d->sock = INVALID_SOCKET;
     config(host, port, ns);
 }
 
 StatsdClient::~StatsdClient()
 {
     // close socket
-    if (d->sock >= 0) {
-#ifdef WIN32
-        int ret = closesocket(d->sock);
-#else
-        int ret = close(d->sock);
-#endif
-        d->sock = -1;
+    if (CloseSocket(d->sock)) {
```

Comment:
I guess this method should practically never return false...but if it does, we leave `d` lingering and never being deleted.